File: practica1/Ejercicio3/src/ejercicio3/EmpleadoMedioTiempoBuilder.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ejercicio3;

/**
 *
 * @author carme
 */
public class EmpleadoMedioTiempoBuilder extends EmpleadoBuilder{
    @Override
    public EmpleadoBuilder build(String nombre, String dni, String cargo) {
        empleado.setNombre(nombre);
        empleado.setDni(dni);
        empleado.setCargo(cargo);
        empleado.setTipoContrato("Medio Tiempo");
        return this;
    }

}

File: practica2/ejercicio_grupal/lib/Model/EmpleadoMedioTiempoBuilder.dart
import 'package:ejercicio_grupal/Model/Departamento.dart';
import 'package:ejercicio_grupal/Model/ElementoEmpresa.dart';
import 'package:ejercicio_grupal/Model/Empleado.dart';
import 'package:ejercicio_grupal/Model/EmpleadoBuilder.dart';

class EmpleadoMedioTiempoBuilder extends EmpleadoBuilder {
  EmpleadoMedioTiempoBuilder(ElementoEmpresa? DepSuperior) : super(DepSuperior);

  @override
  EmpleadoBuilder build(String nombre, String dni, String cargo,
      ElementoEmpresa? DepSuperior, String usuario) {
    empleado = Empleado.vacio();
    empleado.setNombre(nombre);
    empleado.setDni(dni);
    empleado.setCargo(cargo);
    empleado.setTipoContrato("Medio Tiempo");
    empleado.setUsuario(usuario);
    if (DepSuperior != null && DepSuperior is Departamento) {
      empleado.DepSuperior = DepSuperior;
      empleado.DepSuperior?.addElementoEmpresa(empleado);
      empleado.dep_superior = (DepSuperior as Departamento).id;
    } else {
      empleado.DepSuperior = null;
      empleado.dep_superior = null;
    }

    return this;
  }
}
